bob/src/hw_metrics_collector.rs

```diff
@@ -242,52 +294,105 @@ impl DiskStatCollector {
         }
 
         DiskStatCollector {
-            iostat_avl: true,
+            procfs_avl: true,
             disk_metric_data,
+            uptime: DiffContainer::new(),
+            reads: DiffContainer::new(),
+            writes: DiffContainer::new(),
+            read_time: DiffContainer::new(),
+            write_time: DiffContainer::new(),
         }
     }
 
     fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
-        let iostat_lines = self.collect_iostat()?;
-        for i in iostat_lines {
+        let uptime = self.uptime()?;
+        let elapsed = self.uptime.diff(uptime);
+        let diskstats = self.diskstats()?;
+        for i in diskstats {
             let lsp: Vec<&str> = i.split_whitespace().collect();
-            if lsp.len() == 21 {
-                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[0]) {
-                    if let (Ok(rs), Ok(ws)) = (
-                        // readops per s count is in 1st column
-                        Self::parse_iostat_result(lsp[1]),
-                        // writeops per s count is in 7th column
-                        Self::parse_iostat_result(lsp[7]),
+            if lsp.len() >= 8 {
+                // compare device name from 2nd column with disk device names
+                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[2]) {
+                    if let (Ok(r_ios), Ok(w_ios)) = (
+                        // successfull reads count is in 3rd column
+                        lsp[3].parse::<f64>(),
+                        // successfull writes count is in 7th column
+                        lsp[7].parse::<f64>(),
                     ) {
-                        let iops = rs + ws;
+                        let d_r_ios = self.reads.diff(r_ios);
+                        let d_w_ios = self.writes.diff(w_ios);
+                        let iops = (d_r_ios + d_w_ios) / elapsed;
```

Comment:
If `elapsed` is too small, the approximation will be incorrect. I suggest to introduce threshold (500ms), under which the diff is calculated not from the previous value, but from the pre-previous one. And the previous value is also not updated below this threshold. In this way we can guarantee that approximation will be estimated by the meaningful interval (> 500ms). 

Pseudocode:
```
if elapsed < 500 {
    let d_r_ios = r_ios - self.prev_prev_reads;
    let iops = (d_r_ios + d_w_ios) / elapsed_from_prev_prev;
} 
else {
    let d_r_ios = r_ios - self.prev_reads;
    self.prev_prev_reads = self.prev_reads;
    self.prev_reads = r_ios;
    let iops = (d_r_ios + d_w_ios) / elapsed;
}
```